### ISSUE TITLE
[FEATURE REQUEST] Additional tools for masking

### ISSUE THREAD

**Issue by thedyze** [hlky/stable-diffusion-webui#291]
It would be nice to be able to submit an image with a transparent background for inpainting, a la Dalle2.
Alternatively being able to change the masking brush size, or a masking selection box.

**Comment by HenkDz** [comment]
also being able to remove masked objects would be great.

**Comment by hlky** [comment]
The requests regarding the image editor are getting a bit out of scope for this project, you can easily do masking yourself in photoshop or any image editor, that's what most people who are getting good img2img results are doing, or just using it as img2img no masking.

Plus, we're waiting on Gradio to make some changes that will make some of these things possible, it's the best it can be at the moment I'm afraid.

**Comment by thedyze** [comment]
> The requests regarding the image editor are getting a bit out of scope for this project, you can easily do masking yourself in photoshop or any image editor, that's what most people who are getting good img2img results are doing, or just using it as img2img no masking.
> 
> Plus, we're waiting on Gradio to make some changes that will make some of these things possible, it's the best it can be at the moment I'm afraid.

I did not know that was possible, is there anywhere I can read up on that feature?

**Comment by nderscore** [comment]
> I did not know that was possible, is there anywhere I can read up on that feature?

@thedyze 

* Generate a regular image-to-image
* Then some image editor software that supports layers, put the original image on a layer on top of the new one
* Erase what you don't want from the original image, exposing the new image underneath

You have more control over the final result this way too.

**Comment by thedyze** [comment]
> @thedyze
> 
>     * Generate a regular image-to-image
> 
>     * Then some image editor software that supports layers, put the original image on a layer on top of the new one
> 
>     * Erase what you don't want from the original image, exposing the new image underneath
> 
> 
> You have more control over the final result this way too.

Oh, ..yeah I've used that technique. 
I was thinking about inpainting, supply part of an image and leave the rest empty, have the AI fill in the blanks.

**Comment by altryne** [comment]
@thedyze this would get fixed once Gradio introduces their rewamped tools for inpainting. 
see for comments on this #201
